packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -40,15 +40,29 @@ function calculateTwoColumnModuleWidth(columnWidth: number, gutter: number): num
   return columnWidth * 2 + gutter;
 }
 
-function calculateSplitIndex(itemsWithoutPositionsLength: number, twoColumnIndex: number): number {
+function calculateSplitIndex(
+  oneColumnItemsLength: number,
+  twoColumnIndex: number,
+  isFirstRow: ?boolean,
+  fitsFirstRow: ?boolean,
+): number {
+  // We add one more item to pre so it is positioned instead of two column item
+  if (isFirstRow && !fitsFirstRow) {
+    return twoColumnIndex + 1;
```

Comment:
let's pass in availableSlotsOnFirstRow so we make this dynamic
```
if (isFirstRow && !fitsFirstRow) {
  return twoColumnIndex + availableSlotsOnFirstRow;
}
```

Reply:
The correct logic would be:

```
if (isFirstRow && twoColumnIndex < columnCount && !fitsFirstRow) {
  return availableSlotsOnFirstRow;
}
```

since we want the split to be exactly the number of elements we want to insert on the first row